src/limiter/maximum_principle_limiter.h

```diff
@@ -0,0 +1,66 @@
+#ifndef __MAXIMUM_PRINCIPLE_LIMITER__
+#define __MAXIMUM_PRINCIPLE_LIMITER__
+
+#include "bound_preserving_limiter.h"
+
+namespace PHiLiP {
+/**********************************
+* Zhang, Xiangxiong, and Chi-Wang Shu. 
+* "On maximum-principle-satisfying high order schemes for scalar conservation laws." 
+* Journal of Computational Physics 229.9 (2010): 3091-3120.
+**********************************/
+template<int dim, int nstate, typename real>
+class MaximumPrincipleLimiter : public BoundPreservingLimiterState <dim, nstate, real>
+{
+public:
+    /// Constructor
+    explicit MaximumPrincipleLimiter(
+        const Parameters::AllParameters* const parameters_input);
+
+    /// Destructor
+    ~MaximumPrincipleLimiter() = default;
+
+    /// Maximum of initial solution in domain.
+    std::vector<real> global_max;
+    /// Minimum of initial solution in domain.
+    std::vector<real> global_min;
+
+    /// Pointer to TVB limiter class (TVB limiter can be applied in conjunction with this limiter)
+    std::shared_ptr<BoundPreservingLimiter<dim, real>> tvbLimiter;
+
+private:
+    /// Function to obtain the maximum and minimum of the initial solution
+    void get_global_max_and_min_of_solution(
+        const dealii::LinearAlgebra::distributed::Vector<double>&       solution,
+        const dealii::DoFHandler<dim>&                                  dof_handler,
+        const dealii::hp::FECollection<dim>&                            fe_collection);
+
+    /// Function to obtain the solution cell average
+    using BoundPreservingLimiterState<dim, nstate, real>::get_soln_cell_avg;
+    
+    /// Function to verify the limited solution satisfies the strict maximum principle
+    void write_limited_solution(
+        dealii::LinearAlgebra::distributed::Vector<double>&      solution,

```

Comment:
const for solution?

Reply:
I am modifying the solution in this function, should I still have const?

Reply:
Oh then in that case, const would not work since it prevents changes; you can mark this comment as resolved